pkg/stub/handler.go

```diff
@@ -862,6 +960,73 @@ func (h *Handler) processError(opcfg *v1alpha1.SamplesResource, ctype v1alpha1.S
 	return err
 }
 
+func (h *Handler) imageStream(imagestream, is *imagev1.ImageStream, opcfg *v1alpha1.SamplesResource) error {
+	if _, isok := h.skippedImagestreams[imagestream.Name]; !isok {
+		h.updateDockerPullSpec([]string{"docker.io", "registry.redhat.io", "registry.access.redhat.com", "quay.io"}, imagestream, opcfg)
+
+		if imagestream.Labels == nil {
+			imagestream.Labels = make(map[string]string)
+		}
+		imagestream.Labels[v1alpha1.SamplesManagedLabel] = "true"
+		if imagestream.Annotations == nil {
+			imagestream.Annotations = make(map[string]string)
+		}
+		imagestream.Annotations[v1alpha1.SamplesVersionAnnotation] = v1alpha1.CodeLevel
+
+		if is == nil {
+			_, err := h.imageclientwrapper.Create("openshift", imagestream)
+			if err != nil {
+				return h.processError(opcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "imagestream create error: %v")
+			}
+			logrus.Printf("created imagestream %s", imagestream.Name)
+		} else {
+			imagestream.ResourceVersion = is.ResourceVersion
+			_, err := h.imageclientwrapper.Update("openshift", imagestream)
+			if err != nil {
+				return h.processError(opcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "imagestream update error: %v")
+			}
+			logrus.Printf("updated imagestream %s", is.Name)
+		}
+	} else {
+		if is.Labels == nil {
```

Comment:
`is` could still be nil here, presumably.

Reply:
yep ... and same in the template one ... part of next push